app/models/loan.rb

```diff
@@ -16,18 +29,31 @@ class Loan < ApplicationRecord
                                     where(origination_date:
                                       Time.now.beginning_of_year..Time.now)
                                   }
+  # scope :outstanding_invoices, -> { where status: 'outstanding' },
+  #       class_name: 'Invoice',
+  #       foreign_key: 'loan_id'
+
   enum funding_channel: %i[crowdfund sale undefined]
+  enum product: %i[at_close construction_tranche]
   enum status: %i[received under_review term_sheet
                   closing_scheduled closed funded_off_platform
                   fully_funded repaid]
 
-  after_commit :determine_loan_funding_channel, if: :status_changed_to_closed
+  # after_commit :determine_loan_funding_channel, if: :status_changed_to_closed
 
   def determine_loan_funding_channel
     set_loan_funding_channel
     save!
```

Comment:
'.save!' - we rarely want to see a database stuff in the model, that is for the controller, use the model for the business logic / state changes.  Whenever we see database calls in the model, we know there is good chance of n + 1 and difficult to read code.

Reply:
yeah, this smells, I refactored PayoffManager and I think I am going to do something similar with  the `LoanManager` module.